# Task3: Exploratory Data Analysis (EDA)
### By: Priyanka Choudhary

In [1]:
import pandas as pd
import numpy as np

In [2]:
df = pd.read_csv('../data/dataset.csv')
df['Age'].fillna(df['Age'].median(), inplace=True)
df['Embarked'].fillna(df['Embarked'].mode()[0], inplace=True)
df.drop(columns=['Cabin'], inplace=True)
print("Dataset ready! Shape:", df.shape)

Dataset ready! Shape: (891, 11)


In [3]:
df.describe()

,PassengerId,Survived,Pclass,Age,SibSp,Parch,Fare
count,891.000000,891.000000,891.000000,891.000000,891.000000,891.000000,891.000000
mean,446.000000,0.383838,2.308642,29.361582,0.523008,0.381594,32.204208
std,257.353842,0.486592,0.836071,13.019697,1.102743,0.806057,49.693429
min,1.000000,0.000000,1.000000,0.420000,0.000000,0.000000,0.000000
25%,223.500000,0.000000,2.000000,22.000000,0.000000,0.000000,7.910400
50%,446.000000,0.000000,3.000000,28.000000,0.000000,0.000000,14.454200
75%,668.500000,1.000000,3.000000,35.000000,1.000000,0.000000,31.000000
max,891.000000,1.000000,3.000000,80.000000,8.000000,6.000000,512.329200


In [4]:
print("Survival Count:")
print(df['Survived'].value_counts())
print("\nSurvival Percentage:")
print(df['Survived'].value_counts(normalize=True) * 100)

Survival Count:
0    549
1    342
Name: Survived, dtype: int64

Survival Percentage:
0    61.616162
1    38.383838
Name: Survived, dtype: float64


In [5]:
print("Survival by Gender:")
print(df.groupby('Sex')['Survived'].mean() * 100)

Survival by Gender:
Sex
female    74.203822
male      18.890815
Name: Survived, dtype: float64


In [6]:
print("Survival by Passenger Class:")
print(df.groupby('Pclass')['Survived'].mean() * 100)

Survival by Passenger Class:
Pclass
1    62.962963
2    47.282609
3    24.236253
Name: Survived, dtype: float64


In [7]:
print("Age Statistics:")
print("Youngest passenger:", df['Age'].min())
print("Oldest passenger:", df['Age'].max())
print("Average age:", round(df['Age'].mean(), 2))

Age Statistics:
Youngest passenger: 0.42
Oldest passenger: 80.0
Average age: 29.36


In [8]:
print("Correlation with Survival:")
print(df.corr()['Survived'].sort_values(ascending=False))

Correlation with Survival:
Survived       1.000000
Fare           0.257307
Parch          0.081629
PassengerId   -0.005007
SibSp         -0.035322
Age           -0.064910
Pclass        -0.338481
Name: Survived, dtype: float64


## Key Findings
- Only 38% of passengers survived
- Women had 74% survival rate vs men 18%
- 1st class passengers survived more (63%) than 3rd class (24%)
- Youngest passenger was 0.42 years, oldest was 80
- Average age was 29.36 years